crates/re_arrow_store/src/store_polars.rs

```diff
@@ -89,25 +89,48 @@ impl DataStore {
             })
             .collect();
 
+        // Some internal functions of `polars` will panic if everything's empty: early exit.
+        if dfs.iter().all(|df| df.is_empty()) {
+            return DataFrame::empty();
+        }
+
         // Concatenate all indices together.
         //
         // This has to be done diagonally since these indices refer to different entities with
         // potentially wildly different sets of components and lengths.
-        let df = diag_concat_df(dfs.as_slice())
-            // TODO(cmc): is there any way this can fail in this case?
-            .unwrap();
+        //
+        // NOTE: The only way this can fail in this case is if all these frames are empty, because
+        // the store itself is empty, which we check just above.
+        let df = diag_concat_df(dfs.as_slice()).unwrap();
 
+        // Arrange the columns in the order that makes the most sense as a user.
         let df = sort_df_columns(&df, self.config.store_insert_ids);
 
-        let has_insert_ids = df.column(DataStore::insert_id_key().as_str()).is_ok();
-        let has_timeless = df.column(IS_TIMELESS_COL).is_ok();
+        let has_timeless = df.column(TIMELESS_COL).is_ok();
+        let insert_id_col = DataStore::insert_id_key().as_str();
 
+        // Now we want to sort _the contents_ of the columns, and we need to make sure we do so in
+        // as stable a way as possible given our constraints: we cannot actually sort the
+        // component columns themselves as they are internally lists of their own.
         let (sort_cols, sort_orders): (Vec<_>, Vec<_>) = [
-            has_timeless.then_some((IS_TIMELESS_COL, true)),
-            has_insert_ids.then_some((DataStore::insert_id_key().as_str(), false)),
+            df.column(TIMELESS_COL)
+                .is_ok()
+                .then_some((TIMELESS_COL, true)), // descending
+            df.column(insert_id_col)
+                .is_ok()
+                .then_some((insert_id_col, false)), // ascending
         ]
         .into_iter()
         .flatten()
+        // NOTE: Already properly arranged above, and already contains insert_id if needed.
+        .chain(
+            df.get_column_names()
+                .into_iter()
+                .filter(|col| *col != TIMELESS_COL) // we handle this one separately
+                .filter(|col| *col != insert_id_col) // we handle this one separately
+                .filter(|col| !col.starts_with("rerun.")) // lists cannot be sorted
```

Comment:
I don't get this line. `rerun.` is the prefix for official components. Why are they different from `ext.` components? And if it is lists that can't be sorted, why not check whether the column contains lists instead?

Or should this be an allow-list instead of a deny-list (name the columns we sort by, rather the ones we _don't_ sort by)?

Reply:
Yeah I forgot about the `ext.` ones; you're right that a allow list should be better, I'll switch it up